src/libraries/Common/src/Interop/Windows/WinSock/Interop.WSADuplicateSocket.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Net.Sockets;
+using System.Runtime.InteropServices;
+
+internal static partial class Interop
+{
+    internal static partial class Winsock
+    {
+        [StructLayout(LayoutKind.Sequential)]
+        internal struct WSAProtocolChain
+        {
+            internal int ChainLen;
+            [MarshalAs(UnmanagedType.ByValArray, SizeConst=7)]
+            internal uint[] ChainEntries;
+        }
+
+        [StructLayout(LayoutKind.Sequential, CharSet=CharSet.Auto)]
+        internal struct WSAProtocolInfo
+        {
+            internal uint ServiceFlags1;
```

Comment:
All of these should map to the native naming, e.g. dwServiceFlags1, etc.

I'll stop commenting on naming.